src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -239,53 +275,61 @@ protected override void Dispose(bool disposing)
             _link?.Dispose();
         }
 
-        private DateTime? GetTimestamp(string path, IDictionary<string, DateTime> timestampCache)
+        private static DateTime? GetTimestamp(string path, IDictionary<string, DateTime> timestampCache)
         {
             if (!timestampCache.TryGetValue(path, out var time))
             {
                 var info = new FileInfo(path);
-                if (info.Exists)
-                {
-                    time = info.LastWriteTimeUtc;
-                    timestampCache[path] = time;
-                    return time;
-                }
-                else
+                if (!info.Exists)
```

Comment:
The timestamp cache is flushed at the end of the build. It's only for caching across the span of a build operation,